src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/ApplicationInsightsEventPublisher.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.ApplicationInsights;
+using System.Diagnostics;
+
+namespace Microsoft.FeatureManagement.Telemetry.ApplicationInsights
+{
+    /// <summary>
+    /// Listens to <see cref="Activity"/> events from feature management and sends them to Application Insights.
+    /// </summary>
+    internal sealed class ApplicationInsightsEventPublisher : IDisposable
+    {
+        private readonly TelemetryClient _telemetryClient;
+        private readonly ActivityListener _activityListener;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApplicationInsightsEventPublisher"/> class.
+        /// </summary>
+        /// <param name="telemetryClient">The Application Insights telemetry client.</param>
+        public ApplicationInsightsEventPublisher(TelemetryClient telemetryClient)
+        {
+            _telemetryClient = telemetryClient ?? throw new ArgumentNullException(nameof(telemetryClient));
+
+            _activityListener = new ActivityListener
+            {
+                ShouldListenTo = (activitySource) => activitySource.Name == "Microsoft.FeatureManagement",
+                Sample = (ref ActivityCreationOptions<ActivityContext> options) => ActivitySamplingResult.AllData,
+                ActivityStopped = (activity) =>
+                {
+                    ActivityEvent? evaluationEvent = activity.Events.FirstOrDefault((activityEvent) => activityEvent.Name == "feature_flag");
+
+                    if (evaluationEvent != null && evaluationEvent.Value.Tags.Any())
```

Comment:
```suggestion
                    if (evaluationEvent.HasValue && evaluationEvent.Value.Tags.Any())
```

It's functionally the same, that's just the model of nullables.